### PR TITLE
add arguments that allow different optim for G and D; add SGD for optim

### PR DESCRIPTION
**Pull request by AnniLi1212**

### CODE REVIEW

File: setup_training.py [hunk 1]
```diff
@@ -204,7 +204,27 @@ def parse_optimization_args(parser):
         type=str,
         default="rmsprop",
         help="pick optimizer",
-        choices=["adam", "rmsprop", "adadelta", "agcd"],
+        choices=["adam", "rmsprop", "adadelta", "agcd", "sgd"],
+    )
+
+    parser.add_argument(
+        "--use_different_optimizers",
```
Comment (rkansal47) [inline setup_training.py:211]: `--use_different_optimizers` -> `--use-different-optimizers`

File: train.py [hunk 2]
```diff
@@ -23,15 +23,21 @@
...
     device = "cuda" if torch.cuda.is_available() else "cpu"
     torch.autograd.set_detect_anomaly(False)
 
+    args_start_time = time.time()
     args = setup_training.init()
     torch.manual_seed(args.seed)
     args.device = device
-    logging.info("Args initalized")
+    args_end_time = time.time()
+    logging.info("Args initalized in: %s seconds" % (args_end_time - args_start_time))
```
Comment (rkansal47) [inline train.py:40]: Could you reduce the timing statements now? I think some of these will take negligible time (e.g. this one).

**Review by rkansal47 — CHANGES_REQUESTED** [review]

**Comment by rkansal47** [comment]
Looks good! Asked for some minor changes, and reducing the timing statements now.

**Comment by AnniLi1212** [comment]
update profilling, different optim for G and D arg

**Pull request closed without merging**